src/heitech.efXt/IReadRepository.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace heitech.efXt
+{
+    ///<summary>
+    /// A Repository that completely relies on the Specification pattern. If you like a more simple one, use the IRepository
+    ///</summary>
+     public interface IReadRepository<TEntity, TId>
```

Comment:
Does it make sense to constrain the type argument `TId` to implement `IEquatable<TId>` and/or `IComparable<TId>`.
I'm not that familiar with the Entity Framework internals. Do keys need to implement (any of) these interfaces?

Reply:
No it does not, it is only useful for the IRepoistory, as it has a GetbyIdAsync method. Will update